lib/go-tc/servers.go

```diff
@@ -43,12 +47,113 @@ type ServersV1Response struct {
 	Alerts
 }
 
+type ServerDetailV11 struct {
+	ServerDetail
+	LegacyInterfaceDetails
+}
+
+// ServerDetailV30 is the details for a server for API v3
+type ServerDetailV30 struct {
+	ServerDetail
+	ServerInterfaces *[]ServerInterfaceInfo `json:"interfaces"`
+}
+
 // ServersV1DetailResponse is the JSON object returned for a single server for v1.
 type ServersV1DetailResponse struct {
-	Response ServerV1 `json:"response"`
+	Response []ServerDetailV11 `json:"response"`
+	Alerts
+}
+
+// ServersV3DetailResponse is the JSON object returned for a single server for v3.
+type ServersV3DetailResponse struct {
+	Response []ServerDetailV30 `json:"response"`
 	Alerts
 }
 
+// ServerIpAddress is the data associated with a server's interface's IP address.
+type ServerIpAddress struct {
+	Address        string  `json:"address" db:"address"`
+	Gateway        *string `json:"gateway" db:"gateway"`
+	Interface      string  `json:"interface,omitempty" db:"interface"`
+	Server         uint64  `json:"server,omitempty" db:"server"`
```

Comment:
Is it necessary to include these two fields? My `/servers` implementation never used them (and my `ServerIpAddress` struct doesn't have them).